api/naming/naming.go

```
@@ -27,3 +28,17 @@ func Task(t *api.Task) string {
 }
 
 // TODO(stevvooe): Consolidate "Hostname" style validation here.
+
+// Runtime returns the runtime name from a given spec.
+func Runtime(t api.TaskSpec) string {
+	switch r := t.GetRuntime().(type) {
+	case *api.TaskSpec_Attachment:
+		return "attachment"
+	case *api.TaskSpec_Container:
+		return "container"
+	case *api.TaskSpec_Generic:
+		return strings.ToLower(r.Generic.Kind)
+	default:
+		return "none"
```

I would return an error here instead of `"none"`